topics/bloomfilters.md

```diff
@@ -0,0 +1,263 @@
+
+### When should you use scaling vs non-scaling filters
+
+If the capacity (number of items we want to add) is known and fixed, using a non-scaling bloom filter is preferred. Likewise the reverse case, if the capacity is unknown / dynamically calculated, using a scaling bloom filters is ideal.
+
+There are a few benefits for using non scaling filters. A non scaling filter will have better performance than a filter that has scaled out several times (e.g. > 100). Also, non scaling filters in general use less memory for a scaling filter that has scaled out several times to hold the same capacity.
+
+However, to ensure you do not hit any capacity related errors, and want use-as-you-go capacity, scaling is better.
+
+## Bloom properties
+
+* Capacity - The number of unique items that would need to be added before a scale out occurs or (non scaling) before it rejects addition of unique items. 
+
+* False Positive Rate (Error rate) - The rate that controls the probability of bloom check/set operations being false positives. Example: An item addition returning 0 (or an item check returning 1) indicating that the item was already added even though it was not.
+
+* Expansion - This is a property of scalable bloom filters which controls the growth in overall capacity when a bloom filter scales out by determining the capacity of the new sub filter which gets created. This new capacity is equal to the previous filters capacity * expansion rate 
+
+### Advanced Properties
+
+The following two properties can be specified in the `BF.INSERT` command:
+
+* Seed -  This is the key with which hash functions are created for the bloom filter. In case of scalable bloom filters, the same seed is used across all sub filters. This property is only useful if you have a specific 32 byte seed that you want your bloom filter to use. By default every bloom filter will use a random seed.
+
+* Tightening Ratio - This is a property of scalable bloom filters which controls the overall correctness of the bloom filter as it scales out by keeping the actual false positive rate closer to the user requested false positive rate when the bloom filter was created. This is done by using the tightening ratio to set a stricter false positive on the new sub filter which gets created during each scale out. We do not recommend fine tuning this unless there is a specific use case for lower memory usage with higher false positive or vice versa.
+
+### Default bloom properties
+
+These are the default bloom properties along with the commands and configs which allow customizing.
+
+<table width="100%" border="1" style="border-collapse: collapse; border: 1px solid black" cellpadding="8">
```

Comment:
I think this will be wonky when ever we end up putting it in a man page. I think we should keep it as vanilla markdown.